docs/src/advanced-concepts/preparing-generic-targets.rst

```diff
@@ -0,0 +1,110 @@
+Preparing generic targets for reading by metatrain
+==================================================
+
+Besides energy-like targets, the library also supports reading (and training on)
+more generic targets.
+
+Input file
+----------
+
+In order to read a generic target, you will have to specify its layout in the input
+file. Suppose you want to learn a target named ``mtt::my_target``, which is a per-atom
+Cartesian vector with 10 properties. The target section in the input file should look
+like this:
+
+.. code-block:: yaml
+
+    targets:
+      mtt::my_target:
+        quantity: ""
+        read_from: dataset.xyz
+        key: energy
+        unit: ""
+        per_atom: True
+        type:
+          cartiesian:
+            rank: 1
+        num_properties: 10
+
+The crucial fields here are:
+
+- ``per_atom``: This field should be set to ``True`` if the target is a per-atom
+    property. Otherwise, it should be set to ``False``.
+- ``type``: This field specifies the type of the target. In this case, the target is
+    a Cartesian vector. The ``rank`` field specifies the rank of the target. For
+    Cartesian vectors, the rank is 1. Other possibilities are ``scalar`` and
+    ``spherical`` (for a spherical tensor).
+- ``num_properties``: This field specifies the number of different properties in the
+    target that need to be learned. In this case, there are 10 properties.
+
+A few more words should be spent on ``spherical`` targets. These should be made of a
+certain number of irreducible spherical tensors. For example, if you are learning a
+property that can be decomposed into two spherical tensors with L=0 and L=2, the target
+section should would look like this:
+
+.. code-block:: yaml
+
+    targets:
+      mtt::my_target:
+        quantity: ""
+        read_from: dataset.xyz
+        key: energy
+        unit: ""
+        per_atom: True
+        type:
+          spherical:
+            irreps:
+                - o3_lambda: 0
+                  o3_sigma: 1
+                - o3_lambda: 2
+                  o3_sigma: 1
```

Comment:
```suggestion
                - {o3_lambda: 0, o3_sigma: 1}
                - {o3_lambda: 2, o3_sigma: 1}
```

FYI, this is also valid syntax, and might make these things more readable

Reply:
Very good idea